Riot/Modules/Authentication/SSO/SSOAuthentificationSessionProtocol.swift

```diff
@@ -0,0 +1,46 @@
+// 
+// Copyright 2020 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+enum SSOAuthentificationSessionError: Error {
+    case userCanceled
+}
+
+/// A completion handler the session calls when it completes successfully, or when the user cancels the session.
+public typealias SSOAuthentificationSessionCompletionHandler = (URL?, Error?) -> Void
+
+/// An interface the session uses to ask a delegate for a presentation context.
+protocol SSOAuthenticationSessionContextProviding {
```

Comment:
Others named as `Authentification`. Maybe you missed this one. Totally optional.